Revit_Core_Engine/Query/FacadeCurtainPanels.cs

```diff
@@ -0,0 +1,108 @@
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Base;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Facade.Elements;
+using BH.Engine.Facade;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static List<oM.Facade.Elements.Opening> FacadeCurtainPanels(this CurtainGrid curtainGrid, Document document, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null, BH.oM.Adapters.Revit.Enums.Discipline discipline = oM.Adapters.Revit.Enums.Discipline.Physical)
+        {
+            if (curtainGrid == null)
+                return null;
+
+            List<IElement1D> cwMullions = curtainGrid.CurtainWallMullions(document, settings, refObjects).Select(x => x as IElement1D).ToList();
+
+            List<oM.Facade.Elements.Opening> result = new List<oM.Facade.Elements.Opening>();
+            List<Element> panels = curtainGrid.GetPanelIds().Select(x => document.GetElement(x)).ToList();
+            List<CurtainCell> cells = curtainGrid.GetCurtainCells().ToList();
+            if (panels.Count != cells.Count)
+                return null;
+
+            for (int i = 0; i < panels.Count; i++)
+            {
+                Element element = panels[i];
+                Autodesk.Revit.DB.Panel panel = element as Autodesk.Revit.DB.Panel;
+                List<PolyCurve> pcs = new List<PolyCurve>();
+                if (panel == null)
+                { continue; }
+                try
+                { pcs = cells[i].CurveLoops.FromRevit(); }
+                catch
+                { continue; }
```

Comment:
Isn't it worth a warning here? More curious than asking for change.

Reply:
This is a catch for the unique case where a cell exist but the Revit API CurveLoops method throws an exception, which as far as I am aware occurs specifically when the cell has been removed (Ie the wall was shortened and the cell falls off the extents of the wall), hence the warning-less catch. Maybe worth a comment if nothing else though so I have added that for clarity.

Reply:
Comment added in latest commit.